pkg/vdr/web/resolver.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package web
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/did"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+)
+
+// Read resolves a did:web did.
+func (v *VDR) Read(didID string, opts ...vdr.ResolveOpts) (*did.Doc, error) {
+	// apply resolve opts
+	docOpts := &vdr.ResolveDIDOpts{}
+
+	for _, opt := range opts {
+		opt(docOpts)
+	}
+
+	var customClient bool
+
+	var client *http.Client
+
+	switch docOpts.HTTPClient {
+	case nil:
+		client = &http.Client{}
+		customClient = false
+	default:
+		client = docOpts.HTTPClient
+		customClient = true
+	}
+
+	url, err := parseDIDWeb(didID)
+	if err != nil {
+		return nil, fmt.Errorf("error resolving did:web did --> could not parse did:web did --> %w", err)
+	}
+
+	return sendResolveRequest(client, url, customClient)
+}
+
+func sendResolveRequest(client *http.Client, url string, customClient bool) (*did.Doc, error) {
+	resp, err := client.Get(url)
+	if err != nil {
+		switch customClient {
```

Comment:
this code is unnecessarily complicated - it's enough to log error without having to distinguish between default and custom clients.

Reply:
Please include the url in the log message.

Reply:
I agree that it is not necessary to distinguish between default and custom client, but it looks like he really wanted that.